requirements.txt

```diff
@@ -16,7 +16,7 @@ altair==4.2.0
 astroquery==0.4.6
 astropy==4.3.1
 galpy==1.7.2
-catalystcoop.pudl==0.3.2
+catalystcoop.pudl
 dask==2020.12.0
 distributed==2020.12.0
 dask-gateway==0.9.0
```

Comment:
We have pins to ancient versions of distributed and such =/

Reply:
I'd like to propose staying away from conda, and adopting the strategies used by z2jh and binderhub etc where requirements.in + requirements.txt is version controlled, and re-freezes are automated by github actions that can be triggered manually or scheduled.

Reply:
Basically the current setup is untenable, and anything is better than what we have :) I'd actually love to propose retiring this image entirely, and setting up our default to either something from jupyter docker-stacks or pangeo image instead! I do think that for *user images* the community has sort of settled on conda, and I have given up trying to push for freezing purley from pip there - especially after anything touching gdal.

Anyway, I think there are two outcomes here:

1. What to do with *this* repo
2. What to suggest for users who are maintaining their own images

I think for (1) i'd suggest we get rid of it, and move to using the *pangeo images* as default. However, it would need to probably happen during a semester break or something.

Reply:
Trying to get https://pypi.org/project/shapely/ or https://pypi.org/project/websockify/ working was what made me give up on using *purely pip* for user images. The former breaks in funky ways if you try to use it with other packages that use gdal, and the latter installs but doesn't work :D

Reply:
I totally concur with the idea to consume some upstream image as our default image.